main.tf

```diff
@@ -12,20 +12,31 @@ resource "aws_iam_access_key" "default" {
   user = aws_iam_user.default.name
 }
 
-resource "aws_iam_user_policy" "default" {
-  count  = local.create_policy ? 1 : 0
-  name   = "${var.name}${var.postfix ? "Policy" : ""}"
-  user   = aws_iam_user.default.name
-  policy = var.policy
+// Create a default group for user if no groups are specified.
+resource "aws_iam_group" "default" {
+  name = "${var.name}${var.postfix ? "Group" : ""}"
+}
```

Comment:
Lets only create this if needed... And shouldn't we just join any groups supplied with the default group (if needed) and use that instead? Would be much clearer I guess.

Reply:
> If there are groups specified and inline policies are also supplied, we will still create a group with those supplied policies and make created user a member of all groups.